addons/html_editor/static/src/others/embedded_components/plugins/toggle_plugin/toggle_plugin.js

```diff
@@ -0,0 +1,504 @@
+            if (childrenToInsert.length === 1 && isEmptyBlock(childrenToInsert[0])) {
+                childrenToInsert = [];
+            }
+            fragment.replaceChildren(...childrenToInsert);
+            if (fragment.children.length !== 0) {
+                this.dependencies.dom.insert(fragment);
+            }
+            closestToggleTitle
+                .querySelector("p")
+                .insertBefore(
+                    this.document.createTextNode(nextTitle.textContent),
+                    closestToggleTitle.querySelector("p").lastChild
+                );
+            nextSibling.remove();
+            cursors.restore();
+        }
+        return true;
+    }
+
+    /**
+     * Handles all the behaviors linked to the use of deleteBackward in the editor.
+     * We need to handle some specific behaviors:
+     *  1. When we aren't in a toggle title but the previous element is a toggle. (opened and closed)
+     *  2. When we are at the start of the title and we have nothing above the current embedded toggle.
+     * @param {Object} range
+     */
+    handleDeleteBackward(range) {
+        // startContainer should be the editable to indicate the start of the block.
+        const { startContainer, startOffset, endOffset } = range;
+        // endContainer represents the block where the cursor is.
+        const endContainer = closestElement(range.endContainer, isBlock);
+        const closestToggleTitle = closestElement(startContainer, titleSelector);
+        // We are at the start if either we have the same end and start container and the same offset (in title).
+        // Or if the startContainer is the editable with an endOffset of 0 (in the editable).
+        const isCursorAtStart =
+            (closestElement(startContainer) === closestElement(endContainer) &&
+                startOffset === endOffset) ||
+            (startContainer === closestElement(endContainer, "[contenteditable='true']") &&
+                endOffset === 0);
+        if (isCursorAtStart && endContainer.previousElementSibling?.dataset.embedded === "toggle") {
+            // If we are inside the editor after an embedded toggle. We set the cursor to the end
+            // of either the title or the last paragraph of the content.
+            this.dependencies.selection.setCursorEnd(
+                endContainer.previousElementSibling.querySelector(
+                    ".btn:has(.fa-caret-right) + div > [data-embedded-editable='title'] > *, [data-embedded-editable='content'] > p:last-of-type"
+                )
+            );
+            if (!endContainer.textContent) {
+                // If the paragraph we are leaving is empty we remove it to follow the classic
+                // deleteBackwards behavior.
+                endContainer.remove();
+            } else {
+                const { anchorNode } = this.dependencies.selection.getEditableSelection();
+                const { restore: restoreSelection } =
```

Comment:
Why don't we do like in `handleDeleteForward`:  `const cursors = this.dependencies.selection.preserveSelection();` and later `cursors.restore()` ?